playbook/app/pb_kits/playbook/pb_advanced_table/_advanced_table.tsx

```diff
@@ -253,10 +266,117 @@ const AdvancedTable = (props: AdvancedTableProps) => {
   })
 
   const tableRows = table.getRowModel()
+  const tableWrapperRef = useRef<HTMLDivElement>(null)
 
   const hasAnySubRows = tableRows.rows.some(row => row.subRows && row.subRows.length > 0);
   const selectedRowsLength = Object.keys(table.getState().rowSelection).length
 
+  // Generate a flattened set of items for virtualization that includes special components
+  const flattenedItems = useMemo(() => {
+    if (!virtualizedRows) return [];
+
+    const items: Array<{
+      type: 'header' | 'row' | 'loading',
+      row: Row<GenericObject>,
+      id: string
+    }> = [];
+
+    // Process each row and insert special components
+    tableRows.rows.forEach((row) => {
+      const isFirstChildofSubrow = row.depth > 0 && row.index === 0;
+      const subRowHeaders = props.tableOptions?.subRowHeaders;
+
+      if (isFirstChildofSubrow && subRowHeaders) {
+        items.push({
+          type: 'header',
+          row: row,
+          id: `header-${row.id}`,
+        });
+      }
+
+      items.push({
+        type: 'row',
+        row: row,
+        id: row.id
+      });
+
+      const isExpandable = row.getIsExpanded();
+      const rowHasNoChildren = row.original?.children && !row.original.children.length ? true : false;
+      const isDataLoading = isExpandable && (inlineRowLoading && rowHasNoChildren) && 
+        (row.depth < columnDefinitions[0].cellAccessors?.length);
+
+      if (isDataLoading) {
+        items.push({
+          type: 'loading',
+          row: row,
+          id: `loading-${row.id}`
+        });
+      }
+    });
+
+    return items;
+  }, [
+    virtualizedRows,
+    tableRows.rows,
+    props.tableOptions?.subRowHeaders,
+    inlineRowLoading,
+    columnDefinitions
+  ]);
+
+  const tableWrapperStyle = virtualizedRows ? {
+    overflow: 'auto', // scrollable table container
+    position: 'relative', // needed for sticky header
+    height: '600px', // fixed height - adjust as needed
+  } : {}
+
+  // Configure virtualizer to work with our flattened items
+  const virtualizerConfig = useVirtualizer({
+    count: virtualizedRows ? flattenedItems.length : tableRows.rows.length,
+    getScrollElement: () => tableWrapperRef.current,
+    estimateSize: (index) => {
+      if (!virtualizedRows || flattenedItems.length === 0) return 50;
+
+      const item = flattenedItems[index];
+      if (!item) return 50; // Default row height
+
+      if (item.type === 'header') return 40; // Header height
+      if (item.type === 'loading') return 30; // Loading indicator height
+      return 50; // Regular row height
+    },
+    overscan: 5,
+    getItemKey: (index) => virtualizedRows ? flattenedItems[index]?.id || index : tableRows.rows[index]?.id || index,
```

Comment:
@skduncan I noticed we have something similar on the AdvancedTableContext.tsx file. Can we refactor it to reuse the code?